src/cursor/cur_table.c

```diff
@@ -972,7 +972,8 @@ __wt_curtable_open(WT_SESSION_IMPL *session,
 
 	if (0) {
 err:		if (*cursorp != NULL) {
-			WT_TRET(__wt_cursor_close(*cursorp));
+			if (*cursorp != cursor)
+				WT_TRET(__wt_cursor_close(*cursorp));
```

Comment:
@sueloverso The cleanup path is certainly convoluted. Is it true that the only time we'd expect this close call to happen is when opening a dump cursor?